### DISCUSSION TITLE
Templates and Logic in separate files

### DISCUSSION THREAD

**Discussion by go4cas** [vuejs/petite-vue#24] [Q&A]
First off, great lib!

Been looking at the Components section of the docs. I am wondering if it is possible to have the `<template>` and/or js logic of a component in a separate file (either separate for template and logic or combined in a single file), and then import into the main html file.

Or, is this outside of the scope of petite, and more a use case for vue?

**Comment by worldoptimizer** [comment]
As Petite Vue isn't focussed on a render pipeline and has no virtual DOM, I guess this is outside the scope. The template is derived from the in browser DOM, so you could somehow have it inserted there (server side, editor includes or dynamically with JS before).  But I am not entirely sure, … just my current understanding of it.

**Comment by niutech** [comment]
You can make reusable components as separate JS modules with `export function MyComponent(props) { $template: '<div>' + props.name + '</div>' }` and then import them or include components as HTML fragments on server side, e.g. using SSG.
